src/cai_lab_to_nwb/zaki_2024/behaviorinterface.py

```diff
@@ -0,0 +1,67 @@
+"""Primary class for converting experiment-specific behavior."""
+
+from pynwb.file import NWBFile
+
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.datainterfaces import VideoInterface
+from neuroconv.utils import DeepDict
+from pydantic import FilePath
+from typing import Optional
+
+
+class FreezingBehaviorInterface(BaseDataInterface):
+    """Adds intervals of freezing behavior interface."""
+
+    keywords = ["behavior"]
+
+    def __init__(self, file_path: FilePath, video_sampling_frequency: float, verbose: bool = False):
+        # This should load the data lazily and prepare variables you need
+
+        self.file_path = file_path
+        self.verbose = verbose
+        self.video_sampling_frequency = video_sampling_frequency
+
+    def get_metadata(self) -> DeepDict:
+        # Automatically retrieve as much metadata as possible from the source files available
+        metadata = super().get_metadata()
+
+        return metadata
+
+    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: Optional[dict] = None):
+
+        import pandas as pd
+        from ndx_events import Events
+
+        freezing_behavior_df = pd.read_csv(self.file_path)
+
+        # Extract parameters, those values are unique per run
+        file = freezing_behavior_df["File"].unique()[0]
+        motion_cutoff = freezing_behavior_df["MotionCutoff"].unique()[0]
+        freeze_threshold = freezing_behavior_df["FreezeThresh"].unique()[0]
+        min_freeze_duration = freezing_behavior_df["MinFreezeDuration"].unique()[0]
+
+        # Extract timestamps of the freezing events
+        # From the discussion wih the author, the freezing events are the frames where the freezing behavior is 100
+        freezing_frames = freezing_behavior_df[freezing_behavior_df["Freezing"] == 100]["Frame"].values
+        timestamps = freezing_frames / self.video_sampling_frequency
+
+        events_description = (
+            f"Freezing Events Calculated with EzTrack software for file {file} "
+            f"with motion cutoff {motion_cutoff}, freeze threshold {freeze_threshold} "
+            f"and min freeze duration {min_freeze_duration}"
+        )
+
+        events = Events(
+            name="FreezingEvents",
+            description=events_description,
+            timestamps=timestamps,
+        )
+        
```

Comment:
Now that I am looking at this, I think it would be better to use `TimeIntervals` :

- add start and stop times for each freezing and non-freezing period (using the `tags` column to differentiate) 
- associate a timeseries to store the motion variable (using `timeseries` column). 
- I would also add extra columns for `motion_cutoff`, `freeze_threshold` and `min_freeze_duration`, so that we can add detailed description of these parameters.

Reply:
Using `TimeIntervals`
I think this is a good idea.

Some code in that direction:

![image](https://github.com/user-attachments/assets/575e7df2-d94a-4546-937e-16d6e52611e5)

But I am not sure on why we need the non-freezing period. That's like, the normal period right? Having it like that would cover the whole session, right?

Regarding Motion I am not sure, this is like an intermediate variable of Ez-tracking that can be replicated from the video. Moreover, if we go the Intervals route what we would add, the average motion in the interval?

Same for the other columns which do not really change, they are parameters of the algorithm, I don't think it makes sense to add columns for them.

Reply:
> But I am not sure on why we need the non-freezing period. That's like, the normal period right? Having it like that would cover the whole session, right?

> Regarding Motion I am not sure, this is like an intermediate variable of Ez-tracking that can be replicated from the video. Moreover, if we go the Intervals route what we would add, the average motion in the interval?

We add also the non-freezing period to store the timeseries related to the motion. Not the mean but the entire timeseries. There is a specific field for this type of data (timeseries associated to a specific interval) in the TimeInterval table. 
Either this or we  store the MotionTimeSeries in the behavior module and then by looking the start and stop time of freezing or non-freezing intervals, one can have a measurement of motion in that interval by looking at the series

Reply:
OK, I changed this to intervals.

Can you add the MotionSeries in another PR?  I think we would move faster that way and you can choose among any option that you prefer above. Either is good to me.